// passport-facebook/lib/profile.js
/**
 * Parse profile.
 *
 * @param {Object|String} json
 * @return {Object}
 * @api private
 */
exports.parse = function (json) {
    if ('string' == typeof json) {
        json = JSON.parse(json);
    }

    var profile         = {};
    profile.id          = json.id;
    profile.username    = json.username;
    profile.displayName = json.name;
    profile.name        = {
        familyName: json.last_name,
        givenName: json.first_name,
        middleName: json.middle_name
    };

    profile.gender     = json.gender;
    profile.profileUrl = json.link;
    profile.age        = json.age_range.min;

    if(json.languages){
        profile.languages='';
        for (var language in json.languages){
            profile.languages += json.languages[language].name + ","
        }
    }

    if(json.permissions){
        profile.permissions='';
        for (var permission in json.permissions.data){
            profile.permissions += json.permissions.data[permission].permission + ','
        }
    }

    if (json.email) {
        profile.emails = [{value: json.email}];
    }
    ;

    if (json.picture) {
        if (typeof json.picture == 'object' && json.picture.data) {
            // October 2012 Breaking Changes
            profile.photos = [{value: json.picture.data.url}];
        } else {
            profile.photos = [{value: json.picture}];
        }
    }
    ;

    if (json.posts) {
        var postContent = json.posts.data;
        profile.posts = [];
        profile.allPost = '';
        for (var postItem in postContent) {
            if (typeof postContent[postItem].message !== 'undefined') {
                //console.log(postContent[postItem].message);
                profile.allPost += postContent[postItem].message + '\n';
                profile.posts.push({
                    message: postContent[postItem].message,
                    id: postContent[postItem].id,
                    created_time: postContent[postItem].created_time
                });
            }
        }
    } else{
        profile.allPost="";
        profile.posts=[];
    }
    ;
    return profile;
};
